diff --git a/ArduPlane/altitude.cpp b/ArduPlane/altitude.cpp
index 9a0af4c5b5..01de97bc0c 100644
--- a/ArduPlane/altitude.cpp
+++ b/ArduPlane/altitude.cpp
@@ -139,8 +139,14 @@ void Plane::setup_glide_slope(void)
  */
 int32_t Plane::get_RTL_altitude()
 {
+    int32_t rtl_alt;
     if (g.RTL_altitude_cm < 0) {
-        return current_loc.alt;
+        rtl_alt = g.RTL_altitude_cm * -1;
+        if(current_loc.alt >= rtl_alt){
+            return current_loc.alt;
+        } else {
+            return rtl_alt;
+        }
     }
     return g.RTL_altitude_cm + home.alt;
 }
